sdk/eidos_sdk/system/agent_controller.py

```diff
@@ -150,12 +154,41 @@ async def run_and_store_response():
                     raise Exception("Not implemented")
                 return doc
 
-            if execution_mode == "sync":
-                state = await run_and_store_response()
-                return self.doc_to_response(state)
+            if inspect.isasyncgenfunction(handler.fn):
+                async def stream_response():
+                    def to_sse(data: BaseStreamEvent):
+                        return ServerSentEvent(id=str(uuid.uuid4()), data=data.model_dump_json())
```

Comment:
What does the ServerSentEvent do for us? Do we want a uuid on each sse? I would assume we would use our dumped BaseStreamEvent s for the events. EventSourceResponse allows us to just pass through python objects (and presumably the base model as well)